tutorials/ner_pytorch_medical/scripts/torch_ner_model.py

```diff
@@ -0,0 +1,173 @@
+from collections import OrderedDict
+from typing import Optional, List
+from thinc.api import (
+    with_array,
+    chain,
+    Model,
+    PyTorchWrapper,
+    PyTorchLSTM,
+)
+from thinc.types import Floats2d
+
+from spacy.tokens import Doc
+from spacy.util import registry
+import torch
+from torch import nn
+
+
+@registry.architectures("TorchBiLSTMEncoder.v1")
+def BiLSTMEncoder(
+    width: int, depth: int, dropout: float
+) -> Model[List[Floats2d], List[Floats2d]]:
+    """Encode context using bidirectonal LSTM layers. Requires PyTorch.
+    width (int): The input and output width. These are required to be the same,
+        to allow residual connections. This value will be determined by the
+        width of the inputs. Recommended values are between 64 and 300.
+    depth (int): The number of recurrent layers.
+    dropout (float): Creates a Dropout layer on the outputs of each LSTM layer
+        except the last layer. Set to 0 to disable this functionality.
+    """
+    return PyTorchLSTM(width, width, bi=True, depth=depth, dropout=dropout)
```

Comment:
Why do we need this reimplementation of `TorchBiLSTMEncoder`? Does `spacy.TorchBiLSTMEncoder.v1` not work in combination with `spacy.MultiHashEmbed.v2`? Because I thought we [fixed](https://github.com/explosion/spaCy/pull/6442) this? If not - we should definitely look into that again on the `spacy` side of things.

Reply:
Ah right, so I was using spaCy 3.0.5 so I think I needed this for that reason. But this could likely be removed since the project assumes 3.0.6 with those fixes you mention. 

Also, I believe that the current implementation of both `spacy.TorchBiLSTMEncoder.v1` and the thinc PyTorchLSTM have the code
``python
if depth == 0:
    return noop()
```

so that code it's probably unnecessary in the spaCy architecture definition.

Reply:
Also, perhaps the example is clearer if this is included in the project code. The idea here was to say you could keep the spaCy format of embed > encode > attend > predict but for any of those steps you could replace the model with PyTorch. Not sure though

Reply:
Yea I think that would be a great point to make, but the existing spaCy architectures should be compatible. That fix I cited above was already included in 3.0.0, so maybe I have to revisit this again just to be sure.

Reply:
From a [recent forum discussion](https://github.com/explosion/spaCy/discussions/8301#discussioncomment-841740), it seems like the normal built-in `TorchBiLSTMEncoder` should just work. In that case I'd prefer simply using that in the project, because otherwise it seems like the functionality is not readily available in spaCy?

If we want to make the point that another encoder can be swapped in, I feel like it should at least be some encoder that is not currently supported...

Reply:
Could you double check? My little GPU on the laptop is capable of running the `train-trf` command ;-)

Reply:
I tested this and you're right. I must have had something else messed up on my end when I was developing. I'll make that change :)